packages/devextreme/js/__internal/grids/new/grid_core/items_controller/items_controller.ts

```diff
@@ -61,19 +61,24 @@ export class ItemsController {
     const itemKey = this.dataController.getDataKey(data);
 
     return {
+      // @ts-expect-error
       cells: columns.map((column) => {
-        const value = column.calculateCellValue(data);
-        const displayValue = column.calculateDisplayValue(data);
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const formattedText = formatHelper.format(displayValue as any, column.format);
+        const calculatedValue = column.calculateCellValue?.(data);
+        const {
+          // @ts-expect-error
+          column: updatedColumn, value,
+        } = this.columnsController.updateColumnDataType(column, calculatedValue);
```

Comment:
items controller should not modify columns at all
ColumnsController should already contain actual columns configuration